NU1701 - Change the wording of this warning

In a netcoreapp2.0 project, if I add a package that targets net461, I get a new warning as of 2.0 preview2:

> /api/api.csproj : warning NU1701: Package 'Faker.Net 1.0.3' was restored using '.NETFramework,Version=v4.6.1' instead the project target framework '.NETCoreApp,Version=v2.0'. This may cause compatibility problems.

I had a hunch that this was just warning me about the new behavior in 2.0 to reference netfx packages. But I wasn't certain from the message alone. Upon looking up `NU1701` here https://github.com/NuGet/Home/wiki/Restore-errors-and-warnings#nu1701 I found an explanation that I understood: 
> PackageTargetFallback was used to select assets from a package. This is a warning to let the user know that the assets may not be 100% compatible.

Two things helped me verify what the warning was really about: `PackageTargetFallback was used to select assets` and `may not be 100% compatible`

Could we use some of this explanation in the warning? 

Perhaps:

> /api/api.csproj : warning NU1701: Package 'Faker.Net 1.0.3' was restored using '.NETFramework,Version=v4.6.1' instead the project target framework '.NETCoreApp,Version=v2.0'.   This package may not be 100% compatible.

or

> /api/api.csproj : warning NU1701: Package 'Faker.Net 1.0.3' was restored using '.NETFramework,Version=v4.6.1' instead the project target framework '.NETCoreApp,Version=v2.0'.   This package may not be 100% compatible because AssetTargetFallback was used to select assets from this package.

I like the first suggestion. I would avoid the use of `PackageTargetFallback` or `AssetTargetFallback` terms here as that's unnecessary confusion.

Proposed text:
> /api/api.csproj : warning NU1701: Package 'Faker.Net 1.0.3' was restored using '.NETFramework,Version=v4.6.1' instead the project target framework '.NETCoreApp,Version=v2.0'.   This package may not be fully compatible with your project.

Sounds good to me, if you want I can create a PR

-- 



----------
On Wed, Jul 5, 2017 at 7:27 PM, Anand Gaurav <notifications@github.com>
wrote:

I like the first suggestion. I would avoid the use of PackageTargetFallback
or AssetTargetFallback terms here as that's unnecessary confusion.

Proposed text:

/api/api.csproj : warning NU1701: Package 'Faker.Net 1.0.3' was restored
using '.NETFramework,Version=v4.6.1' instead the project target framework
'.NETCoreApp,Version=v2.0'. This package may not be fully compatible with
your project.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/NuGet/Home/issues/5538#issuecomment-313253810>, or mute
the thread
<https://github.com/notifications/unsubscribe-auth/AAK_3bzdTJ4YK6a3jPShwkUceMbGAtWNks5sLBvLgaJpZM4OOgfK>
.

--- a/src/NuGet.Core/NuGet.Commands/Strings.Designer.cs
+++ b/src/NuGet.Core/NuGet.Commands/Strings.Designer.cs
@@ -603,7 +603,7 @@ namespace NuGet.Commands {
         }
         
         /// <summary>
-        ///   Looks up a localized string similar to Package &apos;{0}&apos; was restored using &apos;{1}&apos; instead of the project target framework &apos;{2}&apos;. This may cause compatibility problems..
+        ///   Looks up a localized string similar to Package &apos;{0}&apos; was restored using &apos;{1}&apos; instead of the project target framework &apos;{2}&apos;. This package may not be fully compatible with your project..
         /// </summary>
         internal static string Log_ImportsFallbackWarning {
             get {

--- a/src/NuGet.Core/NuGet.Commands/Strings.resx
+++ b/src/NuGet.Core/NuGet.Commands/Strings.resx
@@ -198,7 +198,7 @@
     <value>Dependency specified was {0} {1} but ended up with {2} {3}.</value>
   </data>
   <data name="Log_ImportsFallbackWarning" xml:space="preserve">
-    <value>Package '{0}' was restored using '{1}' instead of the project target framework '{2}'. This may cause compatibility problems.</value>
+    <value>Package '{0}' was restored using '{1}' instead of the project target framework '{2}'. This package may not be fully compatible with your project.</value>
   </data>
   <data name="Log_CycleDetected" xml:space="preserve">
     <value>Cycle detected.</value>

--- a/test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs
+++ b/test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs
@@ -4509,7 +4509,7 @@ namespace NuGet.CommandLine.Test
                     .ShouldBeEquivalentTo(new[] { "lib/net45/a.dll" },
                     "no compatible assets were found for ns2.0");
 
-                r.AllOutput.Should().Contain("This may cause compatibility problems");
+                r.AllOutput.Should().Contain("This package may not be fully compatible with your project.");
             }
         }
 

--- a/test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommandTests.cs
+++ b/test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommandTests.cs
@@ -746,7 +746,7 @@ namespace NuGet.Commands.FuncTest
                 var lockFileFormat = new LockFileFormat();
                 var command = new RestoreCommand(request);
                 var framework = new FallbackFramework(NuGetFramework.Parse("dotnet"), new List<NuGetFramework> { NuGetFramework.Parse("portable-net452+win81") });
-                var warning = "Package 'Newtonsoft.Json 7.0.1' was restored using '.NETPortable,Version=v0.0,Profile=net452+win81' instead of the project target framework '.NETPlatform,Version=v5.0'. This may cause compatibility problems.";
+                var warning = "Package 'Newtonsoft.Json 7.0.1' was restored using '.NETPortable,Version=v0.0,Profile=net452+win81' instead of the project target framework '.NETPlatform,Version=v5.0'. This package may not be fully compatible with your project.";
 
                 // Act
                 var result = await command.ExecuteAsync();

